webapps/frontend/app/viewmodels/site/vpool/vpool-detail.js

```diff
@@ -117,101 +119,89 @@ define([
             }).promise();
         };
         self.loadStorageRouters = function() {
-            return $.Deferred(function(deferred) {
+            return $.when().then(function () {
                 if (generic.xhrCompleted(self.loadStorageRoutersHandle)) {
                     var options = {
                         sort: 'name',
                         contents: 'storagedrivers,features'
                     };
-                    self.loadStorageRoutersHandle = api.get('storagerouters', { queryparams: options })
-                        .done(function(data) {
+                    self.loadStorageRoutersHandle = StoragerouterService.loadStorageRouterHandle(options)
+                        .done(function (data) {
                             var guids = [], sadata = {};
-                            $.each(data.data, function(index, item) {
+                            $.each(data.data, function (index, item) {
                                 guids.push(item.guid);
                                 sadata[item.guid] = item;
                             });
                             generic.crossFiller(
```

Comment:
Why not use ko.mapping with the storagerouters instead of the crossfiller :thinking: